rime_main.c

```diff
@@ -64,6 +64,112 @@ static void notification_handler(void* context_object,
   }
 }
 
+static int n_plugin_handles;
+static void **plugin_handles;
+static const char **plugin_modules;
+
+static int load_plugins(const char *config_file)
+{
+  RimeConfig config = {0};
+  void **new_plugin_handles;
+
+  if (!rime_api->config_open(config_file, &config)) {
+    g_error("error loading settings for %s\n", config_file);
+    return 1;
+  }
+
+  RimeConfigIterator iter;
+  int n = 0;
+  if (rime_api->config_begin_list(&iter, &config, "plugins")) {
+    while(rime_api->config_next(&iter)) n++;
+    rime_api->config_end(&iter);
+  }
+
+  int m = 0;
+  if (rime_api->config_begin_list(&iter, &config, "modules")) {
+    while(rime_api->config_next(&iter)) m++;
+    rime_api->config_end(&iter);
+  }
+
+  // reserve space new plugins
+  new_plugin_handles = realloc(plugin_handles, sizeof(void *) * (n_plugin_handles + n));
```

Comment:
what happens when `n == 0` and `n_plugin_handles == 0`, that is, when plugin list is empty in config.

Reply:
OK. There is a corner case.
When `n == 0 && n_plugin_handles == 0`, the funtion returns, and the module list is skipped.

I will refactor the code.